packages/admin-panel/src/VizBuilderApp/api/queries/useReportPreview.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import { useQuery } from 'react-query';
+import { post } from '../api';
+import { DEFAULT_REACT_QUERY_OPTIONS } from '../constants';
+
+export const useReportPreview = (config, enabled, setEnabled) =>
+  useQuery(
+    ['fetchReportPreviewData', config],
+    async () => {
+      const response = await post('fetchReportPreviewData', {
+        params: {
+          entityCode: config.location,
+          hierarchy: config.project,
+        },
+        data: { previewConfig: config },
+      });
+
+      return response.results;
+    },
+    {
+      ...DEFAULT_REACT_QUERY_OPTIONS,
+      enabled,
+      keepPreviousData: true,
+      onSettled: () => {
+        setEnabled(false);
```

Comment:
I think there is some refinement that could be done about how the preview interactions work but that probably needs to happen in the user testing. How it works at the moment is that:
- before you've pressed play it shows an 'idle message'
- when you press play it shows the preview table and chart
- when you make changes it simply continues to show the existing state until you press play again